.github/workflows/release.yml

```diff
@@ -2,48 +2,83 @@ name: Run tests and release
 
 on:
   push:
-    tags:
+    tags: "*"
     branches:
       - master
   pull_request:
     branches:
       - master
 
+env:
+  PROJECT_FOLDER: "qgispluginci"
+  PYTHON_VERSION: 3.8
+
 jobs:
-  build:
+  tests:
+    name: "Tests"
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
+    - name: Get source code
+      uses: actions/checkout@v2
 
-    - name: Set up Python 3.7
+    - name: Set up Python
       uses: actions/setup-python@v1
       with:
-        python-version: 3.7
+        python-version: ${{ env.PYTHON_VERSION }}
+
+    - uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/*.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
 
-    - name: Install requirements
+    - name: Install system requirements
       run: |
         sudo apt-get install qt5-default qttools5-dev-tools
-        pip install -r requirements.txt
 
-    - name: Setup
+    - name: Install Python requirements
       run: |
-        VERSION=${GITHUB_REF:-0.0.0}
-        VERSION=${VERSION##*/}
-        sed "s/__VERSION__/${VERSION}/" setup.py.in > setup.py
-        python setup.py install
-        qgis-plugin-ci --version
+        python -m pip install -U pip
+        python -m pip install -r -U requirements.txt
 
     - name: Run tests
       env:
         transifex_token: ${{ secrets.TRANSIFEX_TOKEN }}
         github_token: ${{ secrets.GITHUB_TOKEN }}
       run: nose2 -v
 
+  release:
+    name: "Release on tag 🚀"
+    runs-on: ubuntu-latest
+    needs: [build]
+
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
+
+    steps:
+    - name: Get source code
+      uses: actions/checkout@v2
+
+    - name: Setup
+      run: |
+        VERSION=${GITHUB_REF:-0.0.0}
+        VERSION=${VERSION##*/}
+        sed "s/__VERSION__/${VERSION}/" setup.py.in > setup.py
+
+    - name: Deploy as GitHub Release
```

Comment:
I'd like to avoid this step as I generally create the release via the github website. It's pretty useful, as you usually merge pull request on the website and want to release right after. Does that make sense to you?

Reply:
IMHO we should:

- not rename the setup.* file during the CI but bump the version into the about module (manually as a required "human act" in a release workflow or automatically through something like https://pypi.org/project/bumpversion/
- change the event filter to release create instead of tags OR edit the release after it has been created by the workflow

Reply:
And with a setup.py, we can install in editable mode during dev without gitignoring temporarily

Reply:
> not rename the setup.* file during the CI but bump the version into the about module (manually as a required "human act" in a release workflow or automatically through something like https://pypi.org/project/bumpversion/

What is the main advantage of this?

Reply:
> change the event filter to release create instead of tags OR edit the release after it has been created by the workflow

sounds good!

Reply:
> What is the main advantage of this?

Do you mean the automatic mechanism? or the manual version bumping?

> sounds good!

Which one? :)

Reply:
> Do you mean the automatic mechanism? or the manual version bumping?

Or take this differently, what is the main issue of having a setup.py.in.
If necessary, we could also rename it to setup.py and have version=0.0.0 in it and adapt the sed command.

> Which one? :)
change the event filter to release create instead of tags

Reply:
> > Do you mean the automatic mechanism? or the manual version bumping?
> 
> Or take this differently, what is the main issue of having a setup.py.in.
> If necessary, we could also rename it to setup.py and have version=0.0.0 in it and adapt the sed command.

Completely agree 👍. 

> > Which one? :)
> 
> change the event filter to release create instead of tags

OK.

Reply:
are you on it? I think we'd need to nicely fix the docs build.

Reply:
Done right now (it was breakfast time, sorry :coffee: ):

- fix the doc build (missing ghp-import package)
- remove the create release step
- rename the setup.py.in into setup.py

@3nids not sure about the sed command but I think it would be better to replace it in __about__ and import it into the setup.py, to make it possible to get the version without manipulating the setup.py.

Reply:
that's a very good reason ;)

I am not sure about having it somewhere else. I think we need a valid version number in the setup to run the tests.

I will give you commit rights so you can push to a local branch to have the tests running on your PRs.